### ISSUE TITLE
Error publishing because `networkcloud/v20221212preview` was removed in `@pulumi/azure-native` v2.21.0

### ISSUE THREAD

**Issue by ringods** [pulumi/compliance-policies#40]
### What happened?

Got an error when trying to publish a policy pack using the `azure-native` premium policies:

```sh
$ pulumi policy publish team-ce                   
Obtaining policy metadata from policy plugin
error: [runtime] Running program '/Users/ringods/Projects/pulumi-customers/policy-test' failed with an unhandled exception:

    Error: Cannot find module '@pulumi/azure-native/networkcloud/v20221212preview'

Require stack:

- /Users/ringods/Projects/pulumi-customers/policy-test/node_modules/@pulumi/azure-compliance-policies/azurenative/networkcloud/v20221212preview/index.js
- /Users/ringods/Projects/pulumi-customers/policy-test/node_modules/@pulumi/azure-compliance-policies/azurenative/networkcloud/index.js
- /Users/ringods/Projects/pulumi-customers/policy-test/node_modules/@pulumi/azure-compliance-policies/azurenative/index.js
- /Users/ringods/Projects/pulumi-customers/policy-test/node_modules/@pulumi/azure-compliance-policies/index.js
- /Users/ringods/Projects/pulumi-customers/policy-test/node_modules/@pulumi/compliance-policy-manager/plugin.js
- /Users/ringods/Projects/pulumi-customers/policy-test/node_modules/@pulumi/compliance-policy-manager/index.js
- /Users/ringods/Projects/pulumi-customers/policy-test/index.ts
- /Users/ringods/Projects/pulumi-customers/policy-test/node_modules/@pulumi/pulumi/cmd/run-policy-pack/run.js
- /Users/ringods/Projects/pulumi-customers/policy-test/node_modules/@pulumi/pulumi/cmd/run-policy-pack/index.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Function.Module._load (node:internal/modules/cjs/loader:920:27)
    at Module.require (node:internal/modules/cjs/loader:1141:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (/Users/ringods/Projects/pulumi-customers/policy-test/node_modules/@pulumi/azure-compliance-policies/azurenative/networkcloud/v20221212preview/index.js:24:28)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Function.Module._load (node:internal/modules/cjs/loader:958:12)
    at Module.require (node:internal/modules/cjs/loader:1141:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/ringods/Projects/pulumi-customers/policy-test/node_modules/@pulumi/azure-compliance-policies/azurenative/networkcloud/v20221212preview/index.js',
    '/Users/ringods/Projects/pulumi-customers/policy-test/node_modules/@pulumi/azure-compliance-policies/azurenative/networkcloud/index.js',
    '/Users/ringods/Projects/pulumi-customers/policy-test/node_modules/@pulumi/azure-compliance-policies/azurenative/index.js',
    '/Users/ringods/Projects/pulumi-customers/policy-test/node_modules/@pulumi/azure-compliance-policies/index.js',
    '/Users/ringods/Projects/pulumi-customers/policy-test/node_modules/@pulumi/compliance-policy-manager/plugin.js',
    '/Users/ringods/Projects/pulumi-customers/policy-test/node_modules/@pulumi/compliance-policy-manager/index.js',
    '/Users/ringods/Projects/pulumi-customers/policy-test/index.ts',
    '/Users/ringods/Projects/pulumi-customers/policy-test/node_modules/@pulumi/pulumi/cmd/run-policy-pack/run.js',
    '/Users/ringods/Projects/pulumi-customers/policy-test/node_modules/@pulumi/pulumi/cmd/run-policy-pack/index.js'
  ]
}

error: policy pack "/Users/ringods/Projects/pulumi-customers/policy-test" failed to start: could not read plugin [/opt/homebrew/bin/pulumi-analyzer-policy] stdout: EOF
```

### Example

```sh
$ pulumi policy new azure-iso27001-compliance-policies-typescript
...
$ pulumi policy publish <your-pulumi-org>
```

### Output of `pulumi about`

Of importance is that it breaks with `@pulumi/azure-native` since v2.21.0. See below.

### Additional context

`networkcloud/v20221212preview` was removed in `@pulumi/azure-native` [v2.21.0](https://github.com/pulumi/pulumi-azure-native/releases/tag/v2.21.0):

* https://github.com/pulumi/pulumi-azure-native/tree/v2.20.1/sdk/nodejs/networkcloud
* https://github.com/pulumi/pulumi-azure-native/tree/v2.21.0/sdk/nodejs/networkcloud

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by danielrbradley** [comment]
This specific preview version has been removed by Azure in the specifications:
- https://github.com/Azure/azure-rest-api-specs/pull/26939

As noted by Thomas, running `az provider show --namespace Microsoft.NetworkCloud` only currently shows `2023-07-01` and `2023-05-01-preview`.

Removal of preview versions is fairly common in Azure. What options do we have to prevent failures?

**Comment by aureq** [comment]
fixed in 041a50e1750a150c7b1ff447bd721cae84d45a82
